### PR TITLE
login-signup implementation

### PR DESCRIPTION
**Pull request by Deepika516**
## Description

We created the components Login and signup and deisgn them As per the ARC UI design

Fixes # (issue)
GH-14

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/arc-lib/src/lib/core/auth/auth.service.ts [hunk 1]
```diff
@@ -196,6 +200,45 @@ export class AuthService {
...
+    // const token = new HttpHeaders({
+     
+    //   'Authorization': 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InRpbnlqb2VAZXhhbXBsZS5jb20iLCJpYXQiOjE3MDA2NTk2NTgsImV4cCI6MTcwMDY2MTQ1OCwiaXNzIjoic291cmNlZnVzZSIsInN1YiI6InRpbnlqb2VAZXhhbXBsZS5jb20ifQ.dep1UOseV6XbHRI6SQUUK_kHAqqOOHLXflSyRiogGmA', // Add your authorization token if needed
+    //   // Add other headers as needed
+    // });
+    // command.parameters.headers=headers;
+
+   
+    command.parameters = {
+      data: user,
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/auth/auth.service.ts:219]: Done

File: projects/arc-lib/src/lib/core/auth/models/logged-in-user.model.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export class LoggedInUserDM {
       this.userTenantId = data.userTenantId;
       this.role = data.role;
     } else if (data) {
-      throw new Error('Mandatory fields missing for LoggedInUser model');
+      // throw new Error('Mandatory fields missing for LoggedInUser model');
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/auth/models/logged-in-user.model.ts:55]: i forgot to uncomment this ,anyways Done

File: projects/arc-lib/src/lib/core/interceptors/auth.interceptor.ts [hunk 3]
```diff
@@ -23,11 +24,12 @@ export class AuthInterceptor implements HttpInterceptor {
     req: HttpRequest<unknown>,
     next: HttpHandler
   ): Observable<HttpEvent<unknown>> {
+  
     if (req.headers.has(AuthTokenSkipHeader) || req.url.includes('i18n/')) {
       const headers = req.headers.delete(AuthTokenSkipHeader);
       return next.handle(req.clone({ headers }));
     }
-    const authToken = this.sessionStore.getAccessToken();
+    const authToken = this.sessionStore.getAccessToken() || 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImZlcnJ5QGV4YW1wbGUuY29tIiwiaWF0IjoxNzAxMTgyNDA0LCJleHAiOjE3MDExODQyMDQsImlzcyI6InNvdXJjZWZ1c2UiLCJzdWIiOiJmZXJyeUBleGFtcGxlLmNvbSJ9.ZYq7Y_77_4FYD9YMUKtZQjUc2G1VkpVRy2O1gLPXrIE';
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/interceptors/auth.interceptor.ts:32]: I was checking the sihnup implementation so i hard coded the Token, Anyways Removed

File: projects/arc/src/environments/environment.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-  clientId: '',
-  publicKey: '',
+  clientId: "test_client_id",
+  publicKey: "test_client_secret",
   homePath: '/main/home',
-  baseApiUrl: '',
+  baseApiUrl: 'http://localhost:3002',
   authServiceUrl: '',
-  userServiceUrl: '',
+  userServiceUrl: 'user',
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc/src/environments/environment.ts:8]: Done

File: tsconfig.json [hunk 5]
```diff
@@ -11,7 +11,7 @@
     "moduleResolution": "node",
     "importHelpers": true,
     "target": "ES2019",
-    "typeRoots": ["node_modules/@types"],
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply tsconfig.json:14]: done I add this too

File: projects/arc-lib/src/lib/components/auth/auth.component.ts [hunk 6]
```diff
@@ -8,10 +8,15 @@ import {NbAuthComponent, NbAuthService} from '@nebular/auth';
...
+  altText:string;
   constructor(
     override readonly auth: NbAuthService,
     override readonly location: Location,
   ) {
     super(auth, location);
+    this.image="../../../images/auth/login-img.png";
+    this.altText="loginImg";
+
   }
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/auth.component.ts:21]: removed .i used this in HTML it is not in use RN.

File: projects/arc-lib/src/lib/components/auth/signup/signup.component.scss [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+  background-color: map.get($font, 'light');
+  border: 0.063rem solid map.get($color, 'border-disable');
+  position: absolute;
+
+  &:hover {
+    border-color: map-get($font, 'placeholder');
+  }
+
+  &:active {
+    border-color: map-get($font, 'primary');
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/signup/signup.component.scss:41]: removed

File: projects/arc-lib/src/lib/components/auth/forgot-password/forgot-password.component.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    if (this.forgotPasswordForm.valid) {
+      const email = this.forgotPasswordForm.value.email;
+      this.authService.forgetPasswordReq(email).subscribe(
+        (response) => {
+          // Handle successful link sending
+          console.log('Reset Password link sent successfully:', response);
+        },
+        (error) => {
+          // Handle error
+          console.error('Error sending reset password link:', error);
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/forgot-password/forgot-password.component.ts:45]: Done . i show msg in toaster.

File: projects/arc-lib/src/lib/components/auth/login/login.component.scss [hunk 9]
```diff
@@ -27,15 +29,124 @@
...
+      color: rgb(41, 40, 40);
+      border: 1px solid #8f8f8f;
+    }
+  }
+}
+.sign-in-button {
+  button {
+    font-weight: normal;
+    border: 1px solid #d9d9d9;
+    background: #fff;
```
[the remark this answers is not in the document]
Comment (yeshamavani) [reply projects/arc-lib/src/lib/components/auth/login/login.component.scss]: where is this moved ?

File: projects/arc-lib/src/lib/components/auth/forgot-password/forgot-password.component.ts [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  //         console.log('Reset Password link sent successfully:', response);
+
+  //       },
+  //       (error) => {
+  //         // Handle error
+  //         console.error('Error sending reset password link:', error);
+  //       }
+  //     );
+  //   }
+  // }
```
Comment (yeshamavani) [inline projects/arc-lib/src/lib/components/auth/forgot-password/forgot-password.component.ts:52]: remove commented code from everywhere

File: projects/arc-lib/src/lib/components/auth/reset-password/reset-password.component.scss [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+//   position: absolute;
+
+//   &:hover {
+//     border-color: map-get($font, 'placeholder');
+//   }
+
+//   &:active {
+//     border-color: map-get($font, 'primary');
+//   }
+// }
```
Comment (yeshamavani) [inline projects/arc-lib/src/lib/components/auth/reset-password/reset-password.component.scss:43]: commented code

File: projects/arc-lib/src/lib/core/api/commands/post-api.command.ts [hunk 12]
```diff
@@ -38,7 +38,8 @@ export abstract class PostAPICommand<T extends Partial<R>, R = T>
     return this.apiService
       .post(
         this.uri,
-        this.adapter.adaptFromModel(this.parameters.data),
+      //  this.adapter.adaptFromModel(this.parameters.data),
+      this.parameters.data,
```
[the remark this answers is not in the document]
Comment (yeshamavani) [reply projects/arc-lib/src/lib/core/api/commands/post-api.command.ts:42]: why this change

File: projects/arc-lib/src/lib/core/interceptors/auth.interceptor.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (yeshamavani) [reply projects/arc-lib/src/lib/core/interceptors/auth.interceptor.ts]: while pushing the changes you should never such sensitive information on github or any SCM

**Review by yeshamavani — CHANGES_REQUESTED** [review]
remove all unnecessary code and comments 
all hard coded values, debugger

File: projects/arc-lib/src/lib/components/auth/login/login.component.scss (hunk 9 above)
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/login/login.component.scss]: In global styles.scss

File: projects/arc-lib/src/lib/core/interceptors/auth.interceptor.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/interceptors/auth.interceptor.ts]: ok i ll take care of this now on

File: projects/arc-lib/src/lib/components/auth/forgot-password/forgot-password.component.ts (hunk 10 above)
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/forgot-password/forgot-password.component.ts:52]: ok .. sorry i forgot to remove this one .

File: projects/arc-lib/src/lib/components/auth/reset-password/reset-password.component.scss (hunk 11 above)
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/reset-password/reset-password.component.scss:43]: done

File: projects/arc-lib/src/lib/components/auth/signup/signup.component.ts [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+      //   concatMap(response => {
+          // if (response.body && response.body.code) {
+            this.authService.createExternalUser(userData).subscribe(
+        (resp)=>{
+          // Handle successful login response
+          console.log('signup successful:', resp);
+          this.router.navigate(['/auth/login']);
+        },
+        (error) => {
+          debugger;  // Set a breakpoint here
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/signup/signup.component.ts:53]: done

File: projects/arc-lib/src/lib/core/auth/adapters/index.ts [hunk 14]
```diff
@@ -1,2 +1,3 @@
 export * from './logged-in-user-adapter.service';
 export * from './login-adapter.service';
+// export * from './signup-adapter.service';
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/auth/adapters/index.ts:3]: removed

File: projects/arc-lib/src/lib/components/auth/signup/signup.component.ts [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+  ) {
+    super(route, location);
+    this.imageUrl = '../../../assets/images/auth/ARC_logo.png'; 
+  this.altText = 'logo';
+  this.signupForm = this.fb.group({
+    firstName: ['',[Validators.required]],
+    lastName: ['',Validators.required],
+    email: ['', [Validators.required, Validators.email]],
+    password: ['', [Validators.required, Validators.minLength(6)]],
+  })
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/signup/signup.component.ts:34]: done

File: projects/arc-lib/src/lib/components/auth/login/login.component.scss [hunk 16]
```diff
@@ -19,23 +21,86 @@
...
+      margin-right: 5px;
+    }
   }
 }
+.footer {
+  color: #525252;
+  a {
+    text-decoration: none;
+    color: #19a5ff;
+    font-weight: 700;
```
Comment (yeshamavani) [inline projects/arc-lib/src/lib/components/auth/login/login.component.scss:73]: same why repeating css can we move this to base module/component /auth

File: projects/arc-lib/src/lib/components/auth/signup/signup.component.scss [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+    img{
+        margin-right: 5px;
+    }
+  }
+}
+
+.forgotpassword{
+    position: relative;
+    top: -40px;
+}
```
Comment (yeshamavani) [inline projects/arc-lib/src/lib/components/auth/signup/signup.component.scss:87]: same

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc-lib/src/lib/components/auth/login/login.component.scss (hunk 16 above)
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/login/login.component.scss:73]: i make the same name (.sign-up-button ) and move this code to Main styles.scss file n apply this name to every component where we are using this code

File: projects/arc-lib/src/lib/components/auth/signup/signup.component.scss (hunk 17 above)
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/auth/signup/signup.component.scss:87]: resolved

File: projects/arc-lib/src/lib/components/auth/reset-password/reset-password.component.html [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+  
+        <!-- src="baseHref + 'images/auth/angular.png'"  -->
+      </div>
+      <div class="main-wrapper">
+      <div>
+      <div class="sign-in-title">
+        <h2>Reset Password</h2>
+        <p>
+          Please Enter and confirm your new password .......... for to complete the password reset process
+        </p>
```
[the remark this answers is not in the document]
Comment (yeshamavani) [reply projects/arc-lib/src/lib/components/auth/reset-password/reset-password.component.html:14]: ????

**Comment by yeshamavani** [comment]
css should be in root module and add the styles array in @ngmodule to load the styles 
dont repeat the css in all the components

File: projects/arc-lib/src/lib/core/auth/adapters/signup-adapter.service.ts
Comment (yeshamavani) [inline projects/arc-lib/src/lib/core/auth/adapters/signup-adapter.service.ts:1]: name of the file and class name does not match

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc-lib/src/lib/core/auth/adapters/signup-adapter.service.ts
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/auth/adapters/signup-adapter.service.ts:1]: done

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**